doc/cpp-style-guide.md

```diff
@@ -0,0 +1,902 @@
+
+#### Spaces vs. Tabs
+
+Use only spaces, and indent 2 spaces at a time.
+
+We use spaces for indentation. Do not use tabs in your code. You should set your editor to emit spaces when you hit
+the tab key.
+
+[link to GSG](https://google.github.io/styleguide/cppguide.html#Spaces_vs._Tabs)
+
+#### Function Declarations and Definitions
+
+Return type on the same line as function name, parameters on the same line if they fit. Wrap parameter lists which do not fit on a single line as you would wrap arguments in a function call.
+
+https://google.github.io/styleguide/cppguide.html#Function_Declarations_and_Definitions
+#### Lambda Expressions
+
+Format parameters and bodies as for any other function, and capture lists like other comma-separated lists.
+
+For by-reference captures, do not leave a space between the ampersand (&) and the variable name.
+
+[link to GSG](https://google.github.io/styleguide/cppguide.html#Formatting_Lambda_Expressions)
+
+#### Function Calls
+
+Either write the call all on a single line, wrap the arguments at the parenthesis, or start the arguments on a new line indented by four spaces and continue at that 4 space indent. In the absence of other considerations, use the minimum number of lines, including placing multiple arguments on each line where appropriate.
+
+[link to GSG](https://google.github.io/styleguide/cppguide.html#Function_Calls)
+
+#### Braced Initializer List Format
+
+Format a braced initializer list exactly like you would format a function call in its place.
+
+[link to GSG](https://google.github.io/styleguide/cppguide.html#Braced_Initializer_List_Format)
+
+#### Conditionals
+
+Prefer no spaces inside parentheses. The if and else keywords belong on separate lines.
+
+[link to GSG](https://google.github.io/styleguide/cppguide.html#Conditionals)
+
+#### Loops and Switch Statements
+
+Switch statements may use braces for blocks. Annotate non-trivial fall-through between cases. Braces are optional for single-statement loops. Empty loop bodies should use empty braces or continue.
+
+[link to GSG](https://google.github.io/styleguide/cppguide.html#Loops_and_Switch_Statements)
+
+#### Pointer and Reference Expressions
+
+No spaces around period or arrow. Pointer operators do not have trailing spaces. When declaring a pointer variable or
+argument, place the asterisk adjacent to the type:
+
+```C++
+// These are fine, space following.
+char* c;
+string const& str;
+
+// These are not fine, space preceding.
+char *c;
+string const &str;
+```
+
+[link to GSG](https://google.github.io/styleguide/cppguide.html#Pointer_and_Reference_Expressions)
+
+#### Boolean Expressions
+
+When you have a boolean expression that is longer than the standard line length, be consistent in how you break up
+the lines.  Use the word operators, such as `and`, `not`, and `or`, rather than the punctuation operators such as
+`&&`, `!`, and `||`.  Word operators are more readable.
```

Comment:
*citation needed*

Reply:
No citation.  Neither do any of the other assertions about readability in this style guide or Google's.  And some of the [recommendations](https://engdoc.corp.google.com/eng/doc/devguide/cpp/styleguide.shtml?cl=head#General_Naming_Rules) are actually unsupported by research results:

https://dl.acm.org/citation.cfm?id=3104029

Finally, my experience is that when [shown](https://github.com/GoogleCloudPlatform/google-cloud-cpp/blob/e8726e2838b02ce76ae61e2413d1dda98d4fdf8d/bigtable/client/data.cc#L92) code with the word operators programmers do not even notice, they read the code without problems.

To answer your other question:

> What's the motivation for the naming convention change?

* Consistency with the standard library and Boost.

In addition (and this is far less important):

* The rules are also mechanically enforceable, while at least [one](https://engdoc.corp.google.com/eng/doc/devguide/cpp/styleguide.shtml?cl=head#Function_Names) of the Google rules is not: accessors and modifiers *may* be lowercase, that is bad both because you cannot teach clang-tidy to enforce the rule, and because the implementation of the function changes what names are allowed (start as an accessor, name it lowercase, change the implementation, ooops).